addons/l10n_in_pos/tests/test_in_pos.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+from odoo.addons.account.tests.common import AccountTestInvoicingHttpCommon
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestL10nInPos(AccountTestInvoicingHttpCommon):
+
+    @classmethod
+    @AccountTestInvoicingHttpCommon.setup_chart_template('in')
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company_data['company'].partner_id.update({
+            'country_id': cls.env.ref('base.in').id,
+            'city': 'Gandhinagar',
+        })
+
+        cls.env['product.combo.item'].search([]).unlink()
+        cls.env['product.product'].search([]).write({'available_in_pos': False})
+        cls.product_a = cls.env['product.product'].create({
+            'name': 'Product A',
+            'available_in_pos': True,
+            'l10n_in_hsn_code': '1234',
+            'lst_price': 100.0,
+        })
+        cls.product_b = cls.env['product.product'].create({
+            'name': 'Product B',
+            'available_in_pos': True,
+            'l10n_in_hsn_code': False,
+            'lst_price': 110.0,
+        })
+        cls.cash_pm = cls.env['pos.payment.method'].create({
+            'name': 'Cash',
+            'journal_id': cls.company_data['default_journal_cash'].id,
+            'company_id': cls.company_data['company'].id,
+        })
+        cls.main_pos_config = cls.env['pos.config'].create({
+            'name': 'Shop',
+            'payment_method_ids': [(4, cls.cash_pm.id)],
+        })
```

Comment:
You can use the default test class instead of redefining it here.

See `TestFrontendCommon`